manifests/extract.pp

```diff
@@ -41,25 +41,30 @@
   $strip_components=0,
   $purge=false,
   $user=undef,
-  $tar_command=$::osfamily ? {
-    'Solaris' => 'gtar',
-    default   => 'tar',
-  },
+  $tar_command=undef,
 ) {
 
+  require ::archive::params
```

Comment:
You're mixing up order of parsing and order of evaluation.

Parsing is done on the master side, always top-down.

Evaluation is done on the agent, based on the catalog, and following the dependency tree based on either implicit or explicit dependencies between resources.

The `params` class has no resources at all, only variables, so there's no need for evaluation ordering. Variables will be available if they have been **parsed** before, which is true if you included the class already.